packages/flutter/test/widgets/page_view_test.dart

```diff
@@ -1372,4 +1372,35 @@ void main() {
       await tester.pumpWidget(createPageView(null));
     });
   });
+
+  testWidgets('Get the page value before the content dimension is determined,do not throw an assertion and return null', (WidgetTester tester) async {
```

Comment:
Thanks for adding this test, it looks great!

Small wording suggestion (to hopefully make future debugging easier):

```suggestion
  testWidgets('If ScrollPosition.haveDimensions is false, PageController.page returns null without throwing an assertion', (WidgetTester tester) async {
```